src/Model/Route.php

```diff
@@ -236,7 +226,7 @@ public function getPath()
      * When using PHPCR-ODM, make sure to persist the route before calling this
      * to have the id field initialized.
      */
-    public function setPath($pattern)
+    public function setPath(string $pattern)
```

Comment:
@dbu i had to do that change here to avoid a `null === $pattern` check as `strpos` seems not to like `null` anymore. But this change i also have to make in base interface, wdyt?

Reply:
ahh, this is SF component :-) so we need to change our default values going into the SF component part.
The major issue is: `id` and `idPrefix` is `null` and goes later into an `strpos($id, $idPrefix)`

Reply:
please do a `null === ` check instead of changing the signature. changing that interface is a BC break and its not some kind of internal interface that we expect nobody to have implemented.

for the next major version, its the right thing to make it strict typed (with a changelog note about it), but until then lets not break it.